backend/pkg/api/connect/integration/transform_test.go

```diff
@@ -0,0 +1,259 @@
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusCreated {
+		return fmt.Errorf("unexpected status code: %d message: %s", resp.StatusCode, resp.Status)
+	}
+	return nil
+}
+
+func (s *APISuite) TestDeployTransform() {
+	t := s.T()
+
+	require := rqr.New(t)
+	assert := asrt.New(t)
+
+	t.Run("create transform with valid request (connect-go)", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), transformTimeout)
+		defer cancel()
+
+		topicClient := v1alpha1connect.NewTopicServiceClient(http.DefaultClient, s.httpAddress())
+		require.NoError(spawnTopic(ctx, 2, topicClient, "wasm-tfm-create-test-i"))
+		require.NoError(spawnTopic(ctx, 3, topicClient, "wasm-tfm-create-test-o"))
+
+		transformClient := v1alpha1connect.NewTransformServiceClient(http.DefaultClient, s.httpAddress())
+		stErr := spawnTransform(s.api.Cfg.REST, "test-transform",
+			"wasm-tfm-create-test-i",
+			[]string{"wasm-tfm-create-test-o"},
+			[]adminapi.EnvironmentVariable{{Key: "foo", Value: "bar"}},
+			technicallyATransform)
+
+		assert.NoError(stErr, fmt.Sprintf("error: %v", stErr))
+
+		defer func() {
+			_, err := transformClient.DeleteTransform(ctx, connect.NewRequest(&v1alpha1.DeleteTransformRequest{
+				Name: "test-transform",
+			}))
+			require.NoError(err)
+			require.NoError(despawnTopic(ctx, topicClient, "wasm-tfm-create-test-i"))
+			require.NoError(despawnTopic(ctx, topicClient, "wasm-tfm-create-test-o"))
+		}()
+	})
+}
+
+func (s *APISuite) TestGetTransform() {
+	t := s.T()
+	assert := asrt.New(t)
+	require := rqr.New(t)
+
+	t.Run("create transform with valid request (connect-go)", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), transformTimeout)
+		defer cancel()
+
+		transformClient := v1alpha1connect.NewTransformServiceClient(http.DefaultClient, s.httpAddress())
+		topicClient := v1alpha1connect.NewTopicServiceClient(http.DefaultClient, s.httpAddress())
+		require.NoError(spawnTopic(ctx, 3, topicClient, "wasm-tfm-create-test-c"))
+		require.NoError(spawnTopic(ctx, 2, topicClient, "wasm-tfm-create-test-d"))
+
+		require.NoError(spawnTransform(s.api.Cfg.REST, "test-transform-get", "wasm-tfm-create-test-c", []string{"wasm-tfm-create-test-d"}, []adminapi.EnvironmentVariable{{Key: "foo", Value: "bar"}}, technicallyATransform))
+
+		msg, err := transformClient.GetTransform(ctx, connect.NewRequest(&v1alpha1.GetTransformRequest{
+			Name: "test-transform-get",
+		}))
+		assert.NoError(err)
+
+		assert.Equal(msg.Msg.Transform.Name, "test-transform-get")
+		assert.Equal(msg.Msg.Transform.InputTopicName, "wasm-tfm-create-test-c")
+		assert.Equal(msg.Msg.Transform.OutputTopicNames, []string{"wasm-tfm-create-test-d"})
```

Comment:
expected values go in the first param in asserts, so I think these should be switched around.